algorithms/linfa-elasticnet/src/hyperparams.rs

```diff
@@ -12,19 +12,26 @@ use super::Result;
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-/// A verified hyper-parameter set ready for the estimation of a ElasticNet regression model
-///
-/// See [`ElasticNetParams`](crate::ElasticNetParams) for more informations.
 #[derive(Clone, Debug, PartialEq)]
-pub struct ElasticNetValidParams<F> {
+pub struct ElasticNetValidParamsBase<F, const MULTI_TASK: bool> {
```

Comment:
should the multi task flag not be derived from the dataset?

Reply:
I should be able to encapsulate multi-task use case in another `Fit` impl on the same type

Reply:
I just tried to do that and it didn't work. The single and multitarget `Fit` impls bound the target data type with `AsSingleTarget` and `AsMultiTarget` respectively. If I put both impls on a unified param type then I get a conflicting impl error, even though `AsSingleTarget` and `AsMultiTarget` are implemented on completely different types. This idea would be doable if we made the target types `Array1` and `Array2` instead of generics bounded by traits.

Reply:
the conflicting bounds are probably introduced here: https://github.com/rust-ml/linfa/blob/master/src/dataset/impl_targets.rs#L25-L26 have you also tried bounding the type with `T: AsTargets<Ix = Ix?>` directly?

Reply:
Tried it. Still the same error. Rust compiler probably isn't smart enough to realize that those two bounds are completely disjoint.